### PR TITLE
Fix build with CMake 3.18

### PR DESCRIPTION
**Pull request by TSC21**
Since CMake 3.18, [`find_program()` command _gained a new `REQUIRED` option that will stop processing with an error message if nothing is found_](https://cmake.org/cmake/help/v3.18/release/3.18.html#id5). This was causing the build to fail because the `px4` SITL binary was never found in the first place, since it's built in parallel with `sitl_gazebo`. The `REQUIRED` that was in place wasn't actually doing nothing, because that option didn't exist for versions <=3.17.

I also removed duplicate compiler flags set, which was already being done by `EnableC++XX.cmake`.

### CODE REVIEW

**Review by Jaeyoung-Lim — APPROVED** [review]

**Review by julianoes — COMMENTED** [review]
So we dropped requiring C++14?

**Comment by TSC21** [comment]
> So we dropped requiring C++14?

No: https://github.com/PX4/sitl_gazebo/blob/master/cmake/EnableC%2B%2BXX.cmake

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,8 +20,11 @@ message(STATUS "cmake build type: ${CMAKE_BUILD_TYPE}")
 
 project(mavlink_sitl_gazebo VERSION 1.0.0)
 
-set(CMAKE_CXX_STANDARD 14)
```
Comment (julianoes) [inline CMakeLists.txt:23]: @TSC21 this line is gone?

Comment (TSC21) [reply CMakeLists.txt:23]: Yes because we have https://github.com/PX4/sitl_gazebo/pull/550/files#diff-af3b638bc2a3e6c650974192a53c7291R26.

Comment (julianoes) [reply CMakeLists.txt:23]: Ok, so now we require C++11 or higher.

**Pull request merged**